sklearn/linear_model/coordinate_descent.py

```diff
@@ -1186,8 +1186,9 @@ def fit(self, X, y):
                                          dtype=X.dtype.type)
                 for this_l1_ratio, this_alphas in zip(l1_ratios, alphas)
                 for train, test in folds)
-        mse_paths = Parallel(n_jobs=self.n_jobs, verbose=self.verbose,
-                             **_joblib_parallel_args(prefer="threads"))(jobs)
+        mse_paths = Parallel(
+            n_jobs=self.n_jobs, verbose=self.verbose,
+            **_joblib_parallel_args(require="sharedmem"))(jobs)
```

Comment:
Wait but this will make sure threading is used and allow to change the shared `X` object by different threads.

Is this really what we want? I would have though we wanted to avoid inplace operations in `_path_residuals` (e.g. by triggering a copy)?

Reply:
I don't understand. `prefer='threads'` was already doing that, unless the user specifies another backend, right ?